src/main/java/com/epam/aidial/core/util/BufferingReadStream.java

```diff
@@ -111,7 +118,33 @@ public synchronized ReadStream<Buffer> endHandler(Handler<Void> handler) {
 
     private synchronized void handleChunk(Buffer chunk) {
         content.appendBuffer(chunk);
-        notifyOnChunk(chunk);
+        if (isStreaming && isLastChunk(content)) {
+            lastChunk = chunk;
```

Comment:
We should stop streaming, but still accumulating buffer. At the end we should send the tail. This approach will work for bad response like:
```
data: [DONE]
error: NOT DONE, ha ha
```

Reply:
fixed